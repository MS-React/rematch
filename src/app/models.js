export const game = {
  state: {
    score: 0,
    success: 0,
    fails: 0,
    maxFailures: 5,
    totalToWin: 10,
    timePerProof: 30,
    playing: false,
    player: {
      name: null
    }
  },
  reducers: {
    incrementScore(state, payload) {
      return state.score + payload;
    },
    incrementSuccess(state) {
      state.success + 1;
    },
    incrementFails(state) {
      state.fails + 1;
    },
    proofsLeft(state) {
      state.totalToWin - 1;
    },
    setPlayer(state, payload) {
      state.name = payload;
    },
    pause(state) {
      state.playing = !state.playing;
    }
  }
};
