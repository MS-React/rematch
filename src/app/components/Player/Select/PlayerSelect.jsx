import React from 'react';
import { connect } from 'react-redux';

import './PlayerSelect.scss';

const PlayerSelect = ({ setPlayer }) => (
  <div className="selectPlayer">
    <div>Are you good with Maths? Lets see!</div>
    <div>Choose a Name to start!</div>
    <input type="text" onChange={setPlayer} />
  </div>
);

const mapDispatchToProps = dispatch => ({
  setPlayer: (event) => dispatch.game.setPlayer(event.target.value)
});

export default connect(null, mapDispatchToProps)(PlayerSelect);
