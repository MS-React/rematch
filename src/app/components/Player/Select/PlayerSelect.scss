.selectPlayer {
  display: flex;
  flex-direction: column;
  border: 1px solid #000;
  border-radius: 10px;
  margin: auto;
  padding: 20px;
  width: 300px;
  text-align: center;
}