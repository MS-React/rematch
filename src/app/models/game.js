export default {
  state: {
    score: 0,
    success: 0,
    fails: 0,
    maxFailures: 5,
    totalProofs: 10,
    timePerProof: 30,
    playing: false,
    player: {
      name: null
    },
    proofs: []
  },
  reducers: {
    incrementScore: (state, payload) => ({
      ...state,
      score: state.score + Number(payload)
    }),
    incrementSuccess: (state) => ({
      ...state,
      success: ++state.success
    }),
    incrementFails: (state) => ({
      ...state,
      fails: ++state.fails
    }),
    proofsLeft: (state) => ({
      ...state,
      totalProofs: --state.totalProofs
    }),
    setPlayer: (state, payload) => {
      return {
        ...state,
        player: {
          name: String(payload)
        }
      }
    },
    resumeAndPause: (state) => ({
      ...state,
      playing: !state.playing
    })
  }
};
