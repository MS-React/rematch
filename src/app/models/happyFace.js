export default {
  state: {
    display: false,
    showHappyFace: false,
    showUnhappyFace: false,
  },
  reducers: {
    showHappyFace: state => ({
      ...state,
      display: true,
      showHappyFace: true,
      showUnhappyFace: false,
    }),
    showUnhappyFace: state => ({
      ...state,
      display: true,
      showUnhappyFace: true,
      showHappyFace: false,
    }),
  },
};
