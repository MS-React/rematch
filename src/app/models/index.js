export { default as game } from './game';
export { default as happyFace } from './happyFace';
