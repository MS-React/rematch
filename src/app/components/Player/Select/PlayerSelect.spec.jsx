import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { init } from '@rematch/core'
import * as models from '../../../models'

import PlayerSelect from './PlayerSelect';

describe('PlayerSelect', () => {
  let Component;
  let wrapper;
  let store;

  beforeAll(() => {
    store = init({ models });
  });

  beforeEach(() => {
    Component = shallow(<PlayerSelect store={store} />);
  });

  it('should show the user an input to choose his nickname an a button to start', () => {
    expect(renderer.create(Component.html()).toJSON())
      .toMatchSnapshot();
  });

  describe('setPlayer', () => {
    beforeEach(() => {
      store.dispatch.game.setPlayer = jest.fn();
      wrapper = Component.dive();
      wrapper.find('input').prop('onChange')({
        target: {
          value: 'John Doe'
        }
      });
    });

    it('should dispatch setPlayer from store', () => {
      expect(store.dispatch.game.setPlayer).toHaveBeenCalledWith('John Doe');
    });
  });
});
