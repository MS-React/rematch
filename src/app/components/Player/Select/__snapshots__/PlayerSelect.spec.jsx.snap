// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSelect should show the user an input to choose his nickname an a button to start 1`] = `"<div class=\\"selectPlayer\\"><div>Are you good with Maths? Lets see!</div><div>Choose a Name to start!</div><input type=\\"text\\"/></div>"`;
